clients/admin-ui/src/features/dataset/DatasetCollectionView.tsx

```diff
@@ -0,0 +1,71 @@
+import { Box, Select, Spinner } from "@fidesui/react";
+import { ChangeEvent, useEffect, useState } from "react";
+
+import { FidesKey } from "../common/fides-types";
+import { useGetDatasetByKeyQuery } from "./dataset.slice";
+import DatasetFieldsTable from "./DatasetFieldsTable";
+import { DatasetCollection } from "./types";
+
+const useDataset = (key: FidesKey) => {
+  const { data, isLoading } = useGetDatasetByKeyQuery(key);
+
+  return {
+    isLoading,
+    dataset: data,
+  };
+};
+
+interface Props {
+  fidesKey: FidesKey;
+}
+
+const DatasetCollectionView = ({ fidesKey }: Props) => {
+  const { dataset, isLoading } = useDataset(fidesKey);
+  const [activeCollection, setActiveCollection] = useState<
+    DatasetCollection | undefined
+  >();
+
+  useEffect(() => {
+    if (dataset) {
+      setActiveCollection(dataset.collections[0]);
+    } else {
+      setActiveCollection(undefined);
+    }
+  }, [dataset]);
+
+  if (!dataset) {
+    return <div>Dataset not found</div>;
+  }
+
+  const { collections } = dataset;
+
+  const handleChangeCollection = (event: ChangeEvent<HTMLSelectElement>) => {
+    const collection = collections.filter(
+      (c) => c.name === event.target.value
+    )[0];
+    setActiveCollection(collection);
+  };
+
+  if (isLoading) {
```

Comment:
Should this be moved above before line 36? Could we ever be in a loading state without a dataset?

Reply:
this is one of those cases of "all hooks must render every loop", so putting the `if (isLoading)` above would cut out the useEffect rendering and make react unhappy